src/nbl/video/COpenGL_Queue.h

```diff
@@ -157,22 +197,37 @@ class COpenGL_Queue final : public IGPUQueue
                     auto& submit = p.submit;
 
                     // wait semaphores
-                    // TODO glMemoryBarrier() corresponding to _submit.pWaitDstStageMask[i]
+                    for (uint32_t i = 0; i < submit.waitSemaphoreCount; ++i)
+                    {
+                        GLbitfield barrierBits = pipelineStageFlagsToMemBarrierBits(submit.pWaitDstStageMask[i]);
+                        gl.glSync.pglMemoryBarrier(barrierBits);
```

Comment:
first do a loop and bitwise OR the barrier bits, no point doing multiple barriers in OpenGL!

Reply:
hmm I might have misunderstood how the Vulkan memory model works
https://discord.com/channels/427551838099996672/427803486269210627/871146475256676393